### ISSUE TITLE
Convert to TypeScript

### ISSUE THREAD

**Issue by tony** [isaul32/ckeditor5-math#99]
With CKEditor5 v37 being in 100% TypeScript, maybe we should make the move.

Thoughts @isaul32? I can make a PR

**Comment by tony** [comment]
@isaul32 Any objections with me porting the package to TypeScript?

**Comment by isaul32** [comment]
Sure, if you have the motivation to do it.

**Comment by tony** [comment]
@isaul32 Thank you! I may in the coming weeks

**Comment by xcdmrCHP** [comment]
When the file was "src\ckeditor.js", I could successfully build the project. However, now that it has been changed to "src\ckeditor.ts", I am encountering build errors.And I don't know ts.

```
E:\桌面\ckeditor5-41.0.0-9chhinycvdhp>npm run build

> ckeditor5-custom-build@0.0.1 build
> webpack --mode production

assets by status 2.62 MiB [cached] 71 assets
orphan modules 5.11 MiB [orphan] 1574 modules
runtime modules 2.06 KiB 6 modules
cacheable modules 5.12 MiB
  modules by path ./node_modules/@ckeditor/ 437 KiB 83 modules
  modules by path ./node_modules/color-convert/*.js 20.5 KiB
    ./node_modules/color-convert/index.js 1.67 KiB [built] [code generated]
    ./node_modules/color-convert/conversions.js 16.6 KiB [built] [code generated]
    ./node_modules/color-convert/route.js 2.2 KiB [built] [code generated]
  modules by path ./node_modules/css-loader/dist/runtime/*.js 3.84 KiB
    ./node_modules/css-loader/dist/runtime/cssWithMappingToString.js 2.27 KiB [built] [code generated]
    ./node_modules/css-loader/dist/runtime/api.js 1.57 KiB [built] [code generated]
  ./src/ckeditor.ts + 1093 modules 4.66 MiB [built] [code generated]
  ./node_modules/style-loader/dist/runtime/injectStylesIntoStyleTag.js 6.67 KiB [built] [code generated]
  ./node_modules/color-name/index.js 4.51 KiB [built] [code generated]

ERROR in E:\桌面\ckeditor5-41.0.0-9chhinycvdhp\src\ckeditor.ts
60:17-51
[tsl] ERROR in E:\桌面\ckeditor5-41.0.0-9chhinycvdhp\src\ckeditor.ts(60,18)
      TS7016: Could not find a declaration file for module '@isaul32/ckeditor5-math/src/math'. 'E:\桌面\ckeditor5-41.0.0-9chhinycvdhp\node_modules\@isaul32\ckeditor5-math\src\math.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/isaul32__ckeditor5-math` if it exists or add a new declaration (.d.ts) file containing `declare module '@isaul32/ckeditor5-math/src/math';`
ts-loader-default_e3b0c44298fc1c14

ERROR in E:\桌面\ckeditor5-41.0.0-9chhinycvdhp\src\ckeditor.ts
117:2-13
[tsl] ERROR in E:\桌面\ckeditor5-41.0.0-9chhinycvdhp\src\ckeditor.ts(117,3)
      TS2304: Cannot find name 'Mathematics'.
ts-loader-default_e3b0c44298fc1c14

webpack 5.90.1 compiled with 2 errors in 12631 ms
```

**Comment by fedemp** [comment]
Hi, pals. FYI I'm doing this right now.

**Comment by tony** [comment]
@fedemp You're converting the package to TypeScript?! (If so, go ahead, happy to QA!)

**Comment by fedemp** [comment]
@tony Currently a WIP at https://github.com/fedemp/ckeditor5-math but as mentioned in the linked issue, TS has found some issues.

**Comment by tony** [comment]
@fedemp Solid. 

>  TS has found some issues.

Yep, that is to be expected: typings will need to be added everywhere. 

The effort involved is pretty big - even to someone being familiar with TypeScript and CKEditor. Keep it up if you want to, if you have PR I'm happy to QA. If you want to pause and don't mind someone else trying, you can let the thread know here.

**Comment by fedemp** [comment]
There's a draft pull request at https://github.com/isaul32/ckeditor5-math/pull/127

I still have to make tests work.

**Comment by tony** [comment]
#130

**Comment by tony** [comment]
Fixed by @fedemp via [3b57dbf](https://github.com/isaul32/ckeditor5-math/commit/3b57dbf36d79ba671dc31f739360dbedd1c154e2) in #130.
